language-plutus-core/src/Language/PlutusCore/Universe/Core.hs

```diff
@@ -0,0 +1,348 @@
+    showsPrec pr (AG a) = gshowsPrec pr a
+
+instance GShow f => Show (Some f) where
+    showsPrec pr (Some a) = ($(makeShowsPrec ''Some)) pr (Some (AG a))
+
+instance GShow uni => GShow (In uni) where gshowsPrec = showsPrec
+instance GShow uni => Show (In uni a) where
+    showsPrec pr (In uni) = ($(makeShowsPrec ''In)) pr (In (AG uni))
+
+instance (GShow uni, Closed uni, uni `Everywhere` Show) => GShow (Of uni) where gshowsPrec = showsPrec
+instance (GShow uni, Closed uni, uni `Everywhere` Show) => Show (Of uni a) where
+    showsPrec pr (Of uni x) = bring (Proxy @Show) uni $ ($(makeShowsPrec ''Of)) pr (Of (AG uni) x)
+
+-------------------- 'Pretty'
+
+instance GShow uni => Pretty (In uni a) where
+    pretty (In uni) = pretty $ gshow uni
+
+instance (Closed uni, uni `Everywhere` Pretty) => Pretty (Of uni a) where
+    pretty (Of uni x) = bring (Proxy @Pretty) uni $ pretty x
+
+instance GShow uni => Pretty (Some (In uni)) where
+    pretty (Some s) = pretty s
+
+instance (Closed uni, uni `Everywhere` Pretty) => Pretty (Some (Of uni)) where
+    pretty (Some s) = pretty s
+
+-------------------- 'Eq' / 'GEq'
+
+instance GEq f => Eq (Some f) where
+    Some a1 == Some a2 = a1 `defaultEq` a2
+
+deriving newtype instance GEq uni => GEq (In uni)
+
+instance (GEq uni, Closed uni, uni `Everywhere` Eq) => GEq (Of uni) where
+    Of uni1 x1 `geq` Of uni2 x2 = do
+        Refl <- uni1 `geq` uni2
+        guard $ bring (Proxy @Eq) uni1 (x1 == x2)
+        Just Refl
+
+instance GEq uni => Eq (In uni a) where
+    a1 == a2 = a1 `defaultEq` a2
+
+instance (GEq uni, Closed uni, uni `Everywhere` Eq) => Eq (Of uni a) where
+    a1 == a2 = a1 `defaultEq` a2
+
+-------------------- 'NFData'
+
+instance NFData (In uni a) where
+    rnf (In uni) = uni `seq` ()
+
+instance (Closed uni, uni `Everywhere` NFData) => NFData (Of uni a) where
+    rnf (Of uni x) = bring (Proxy @NFData) uni $ rnf x
+
+instance NFData (Some (In uni)) where
+    rnf (Some s) = rnf s
+
+instance (Closed uni, uni `Everywhere` NFData) => NFData (Some (Of uni)) where
+    rnf (Some s) = rnf s
+
+-------------------- 'Hashable'
+
+instance Closed uni => Hashable (In uni a) where
+    hashWithSalt salt (In uni) = hashWithSalt salt $ tagOf uni
+
+instance (Closed uni, uni `Everywhere` Hashable) => Hashable (Of uni a) where
+    hashWithSalt salt (Of uni x) =
+        bring (Proxy @Hashable) uni $ hashWithSalt salt (Some (In uni), x)
+
+instance Closed uni => Hashable (Some (In uni)) where
+    hashWithSalt salt (Some s) = hashWithSalt salt s
+
+instance (Closed uni, uni `Everywhere` Hashable) => Hashable (Some (Of uni)) where
+    hashWithSalt salt (Some s) = hashWithSalt salt s
+
+-------------------- 'Lift'
+
+-- | 'GLift' to 'Lift' is what 'GShow' to 'Show'.
+class GLift f where
+    glift :: f a -> Q Exp
+    default glift :: Lift (f a) => f a -> Q Exp
+    glift = lift
+
+instance GLift f => Lift (AG f a) where
+    lift (AG a) = glift a
+
+-- See Note [Lift, the black sheep].
```

Comment:
Doesn't exist, I guess you merged the note into the big one.

Reply:
Yep, thanks (it turned out there was an entire species of black sheep).